### ISSUE TITLE
State tags use different styling in the campaigns section

### ISSUE THREAD

**Issue by clarkepaul** [silverstripe/silverstripe-campaign-admin#77]
The badges for draft/modified in the campaigns sections use an older style and need to updated to reflect the style without a background as per gridfields.

If there isn't a reusable style for the newer style one should be created so it can be used throughout the CMS.

Was:
![image](https://user-images.githubusercontent.com/555033/32476121-542756f6-c3da-11e7-939d-bc92e55a4a6c.png)

Should be:
![image](https://user-images.githubusercontent.com/555033/32476137-65c3f720-c3da-11e7-8c84-52fab7e3f10e.png)

**Comment by robbieaverill** [comment]
@clarkepaul here's the badge variations: https://silverstripe.github.io/silverstripe-admin/?knob-message=Hello%20World&knob-status=default&selectedKind=Admin%2FBadges&selectedStory=Badge&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

I think what you're referring to here is a legacy badge style - the ones in the storybook are React badge component states. We may need to add a new one that has a transparent background while still using the same text colours before we can implement this in campaign-admin - thoughts?

**Comment by robbieaverill** [comment]
PR at #817 for the transparent badges. I think the orange used in the screenshot is different from the badges in the pattern library, so we may need to add a new state, or update the "warning" state to use that colour instead

**Comment by clarkepaul** [comment]
Yeah having a transparent set would work well so cheers!

The way I see it, we would be able to set the background as either: no class (slightly rounded square) , badge-pill, or badge-transparent (I'd imagine in Storybook there is a radio to toggle between the three options if possible for testing). 

.badge-transparent could then be used to start replacing all of the older status-flag's through out the CMS. Not sure if you've done this @robbieaverill but can the transparent version have no padding left and right please?
I can review the orange colour separately as that more of a accessibility concern.

**Comment by clarkepaul** [comment]
Bootstrap already uses different left/right padding for the badge-pills, for me this is just another variant of that. The reason for no padding left and right is that it needs to align with other text when used on a new line. 
For example 
![image](https://user-images.githubusercontent.com/555033/52236196-0f632880-292b-11e9-9c61-05231a94a5a2.png)
I don't like using the padding as a separate modifier as you have more chance of breaking the other badges, and it only needs to go with the transparent background. Instead of using "Transparent" which isn't really a good class name, could you use badge-state? as it is going to be used to display the state of a page.

**Comment by robbieaverill** [comment]
So I'm going to update the PR at https://github.com/silverstripe/silverstripe-admin/pull/817 to use `stateBadge` as a boolean modifier instead of `transparent`. Example usage:

* `<Badge message="PUBLISHED" status="success" inverted stateBadge  />` - published
* `<Badge message="MODIFIED" status="warning" inverted stateBadge  />` - modified

(example code only, colours may need adjusting)

**Comment by robbieaverill** [comment]
I've made a new PR which adds a new versioned badge component, essentially a lightweight wrapper around Badge with some specific styles for this use case. See https://github.com/silverstripe/silverstripe-admin/pull/861

**Comment by robbieaverill** [comment]
PR at https://github.com/silverstripe/silverstripe-campaign-admin/pull/140

**Comment by robbieaverill** [comment]
Admin PR is merged, so we can now implement the new `VersionedBadge` component in campaign-admin for SS 4.5

**Comment by GuySartorelli** [comment]
No longer relevant
